frontend/src/metabase/nav/components/SearchResults.tsx

```diff
@@ -44,19 +45,17 @@ export const SearchResults = ({
     [searchText],
   );
 
-  const query = useMemo(
-    () => ({
-      q: debouncedSearchText,
-      limit: DEFAULT_SEARCH_LIMIT,
-      ...searchFilters,
-      models: models ?? searchFilters.type,
-    }),
-    [debouncedSearchText, searchFilters, models],
-  );
+  const query = {
+    q: debouncedSearchText,
+    limit: DEFAULT_SEARCH_LIMIT,
+    ...searchFilters,
+    models: models ?? searchFilters.type,
+  };
 
   const { data: list = [], isLoading } = useSearchListQuery({
     query,
     reload: true,
+    enabled: !!debouncedSearchText,
```

Comment:
We don't want to run a search if we don't have text (without this, we trigger a `/search?limit=50` query